submission/notifications/src/main/kotlin/ebi/ac/uk/notifications/service/RtNotificationService.kt

```diff
@@ -0,0 +1,59 @@
+package ebi.ac.uk.notifications.service
+
+import ebi.ac.uk.extended.model.ExtSubmission
+import ebi.ac.uk.notifications.api.RtClient
+import ebi.ac.uk.notifications.integration.templates.SuccessfulSubmissionModel
+import ebi.ac.uk.notifications.integration.templates.SuccessfulSubmissionTemplate
+import ebi.ac.uk.notifications.persistence.service.NotificationPersistenceService
+import ebi.ac.uk.util.date.toStringDate
+import org.apache.commons.io.IOUtils
+import org.springframework.core.io.ResourceLoader
+import java.nio.charset.Charset
+
+internal const val FROM = "biostudies@ebi.ac.uk"
+internal const val SUCCESSFUL_SUBMISSION_TEMPLATE = "successful-submission.txt"
+internal const val SUCCESSFUL_RESUBMISSION_TEMPLATE = "successful-resubmission.txt"
+
+class RtNotificationService(
+    private val rtClient: RtClient,
+    private val resourceLoader: ResourceLoader,
+    private val notificationPersistenceService: NotificationPersistenceService
+) {
+    fun notifySuccessfulSubmission(submission: ExtSubmission, ownerFullName: String, uiUrl: String) {
+        val accNo = submission.accNo
+        val subject = "BioStudies Successful Submission - $accNo"
+        val notification = submissionNotification(submission, ownerFullName, uiUrl)
+        val content = SuccessfulSubmissionTemplate(templateContent(submission)).getContent(notification)
+
+        when (val ticketId = notificationPersistenceService.findTicketId(accNo)) {
+            null -> createTicket(accNo, subject, submission.owner, content)
+            else -> rtClient.commentTicket(ticketId, content)
+        }
+    }
+
+    private fun createTicket(accNo: String, subject: String, owner: String, content: String) {
+        val ticketId = rtClient.createTicket(accNo, subject, owner, content)
+        notificationPersistenceService.saveRtNotification(accNo, ticketId)
+    }
+
+    private fun templateContent(submission: ExtSubmission): String {
+        val template = when (submission.version) {
+            1 -> SUCCESSFUL_SUBMISSION_TEMPLATE
```

Comment:
this looks like let escape some of the business knowledge of ext submission, maybe we should add extended property isNew or is aReSubmission.

Still not sure this is a 100% good recomendation so ignore if you do not agree.

Reply:
This would require some other changes in several other places making this PR even bigger. I think we could do this in a separate PR.